libs/runtime/ebpf_epoch.c

```diff
@@ -358,7 +364,7 @@ ebpf_epoch_exit(_In_ ebpf_epoch_state_t* epoch_state)
     if (cpu_id != epoch_state->cpu_id) {
         // Assert that the IRQL is < DISPATCH_LEVEL. If it is DISPATCH_LEVEL, then the thread moved to a different CPU
         // (by dropping below DISPATCH_LEVEL) and calling ebpf_epoch_exit(). This is not allowed.
-        ebpf_assert(epoch_state->irql_at_enter < DISPATCH_LEVEL);
+        EBPF_EPOCH_FAIL_FAST(FAST_FAIL_INVALID_ARG, epoch_state->irql_at_enter < DISPATCH_LEVEL);
```

Comment:
This still looks like it's decreasing perf in the hot path

Reply:
This is a cold path. This path is at passive IRQL and involves sending a DPC to another CPU and waiting for it to complete. 

It should be hit relatively rarely and only when a thread migrates between CPUs during an epoch. If this becomes a hot path, we need to rethink the design.